builtInFunctions/esdtNFTTransfer.go

```diff
@@ -343,23 +343,25 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 	return vmOutput, nil
 }
 
-func hasDynamicRole(account vmcommon.UserAccountHandler, tokenID []byte, marshaller vmcommon.Marshalizer) (bool, error) {
+func hasDynamicRole(account vmcommon.UserAccountHandler, tokenID []byte, marshaller vmcommon.Marshalizer, enableEpochsHandler vmcommon.EnableEpochsHandler) (bool, error) {
 	roleKey := append(roleKeyPrefix, tokenID...)
 	roles, _, err := getESDTRolesForAcnt(marshaller, account, roleKey)
 	if err != nil {
 		return false, err
 	}
 
 	dynamicRoles := [][]byte{
-		[]byte(core.ESDTRoleNFTAddURI),
-		[]byte(core.ESDTRoleNFTUpdateAttributes),
 		[]byte(core.ESDTMetaDataRecreate),
```

Comment:
these roles have been newly added. No need for enableEpochsHandler for them?

Reply:
No, they will be added to an address only after the activation of the `DynamicNFTFlag`